tests/st/calicoctl/test_crud.py

```diff
@@ -565,6 +569,109 @@ def test_clusterinfo(self):
+        """
+        Test that the metadata fields other than labels and annotations cannot be changed
+        in create and update operations by applying a resource twice.
+        """
+        # Use apply to create a new Host Endpoint and get it to determine the
+        # current resource version (first checking that it doesn't exist).
+        rc = calicoctl(
+            "get hostendpoint %s -o yaml" % name(hostendpoint_name1_rev2))
+        rc.assert_error(text=NOT_FOUND)
+
+        rc = calicoctl("apply", data=hostendpoint_name1_rev2)
+        rc.assert_no_error()
+        rc = calicoctl(
+            "get hostendpoint %s -o yaml" % name(hostendpoint_name1_rev2))
+        rc.assert_no_error()
+        rev0 = rc.decoded
+        self.assertEqual('uid' in rev0['metadata'], True)
+        self.assertEqual('creationTimestamp' in rev0['metadata'], True)
+
+        # Apply the Host Endpoint (with no resource version) and get it to
+        # assert the resource version is not the same.
+        rc = calicoctl("apply", data=hostendpoint_name1_rev3)
+        rc.assert_no_error()
+        rc = calicoctl(
+            "get hostendpoint %s -o yaml" % name(hostendpoint_name1_rev3))
+        rc.assert_no_error()
+        rev1 = rc.decoded
+        self.assertNotEqual(rev0['metadata']['resourceVersion'], rev1['metadata']['resourceVersion'])
+
+        # Validate that only annotations and labels were changed in the metadata
+        self.assertEqual('selfLink' in rev1['metadata'], False)
+        self.assertEqual('generation' in rev1['metadata'], False)
+        self.assertEqual('finalizers' in rev1['metadata'], False)
+        self.assertEqual('uid' in rev1['metadata'], True)
+        self.assertEqual('creationTimestamp' in rev1['metadata'], True)
+        self.assertNotEqual(rev1['metadata']['uid'], hostendpoint_name1_rev3['metadata']['uid'])
+        self.assertNotEqual(rev1['metadata']['creationTimestamp'], hostendpoint_name1_rev3['metadata']['creationTimestamp'])
+        self.assertEqual(rev1['metadata'].get('name', ''), hostendpoint_name1_rev2['metadata'].get('name', ''))
+        self.assertEqual(rev1['metadata']['labels'], hostendpoint_name1_rev3['metadata']['labels'])
+        self.assertNotEqual(rev1['metadata']['labels'], rev0['metadata']['labels'])
+        self.assertEqual(rev1['metadata']['annotations'], hostendpoint_name1_rev3['metadata']['annotations'])
+        self.assertNotEqual(rev1['metadata']['annotations'], rev0['metadata']['labels'])
+
+        # Validate that creationTimestamp and UID are unchanged from when they were created
+        self.assertEqual(rev1['metadata']['creationTimestamp'], rev0['metadata']['creationTimestamp'])
+        self.assertEqual(rev1['metadata']['uid'], rev0['metadata']['uid'])
+
+        # Delete the resource without using a resource version.
+        rc = calicoctl("delete hostendpoint %s" % name(rev1))
+        rc.assert_no_error()
+
+    def test_metadata_unchanged_create_apply(self):
```

Comment:
I was thinking one this one would be create/replace rather than using apply.

Reply:
hmmm, in k8s - does a `replace` act as if the resource is a completely new object with a new UUID and new creation timestamp?